### PR TITLE
WT-4344 Add a set of tests for salvaging corrupt turtle files

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: test/suite/test_txn19.py [hunk 1]
```diff
@@ -346,12 +346,120 @@ def test_corrupt_log(self):
...
         self.reopen_conn(newdir2, self.conn_config)
         self.checks(expect)
 
+class test_txn19_meta(wttest.WiredTigerTestCase, suite_subprocess):
+    base_config = 'log=(archive=false,enabled,file_max=100K),' + \
+                  'transaction_sync=(enabled,method=none)'
+    conn_config = base_config
+
+    # The type of corruption to be applied
+    corruption_scen = [
```
Comment (luke-pearson) [inline test/suite/test_txn19.py]: Would prefer to call it corruption_scenarios so we are consistent in this file. But happy to ignore.

Comment (ddanderson) [reply test/suite/test_txn19.py]: Fixed this, thanks!

**Comment by ddanderson** [comment]
retest this please

**Comment by ddanderson** [comment]
@luke-pearson, thanks for the review.  I added some more complete testing for the meta file cases based on some things mentioned in the JIRA ticket, and wrote up my "results" in the ticket. I'll spawn a few new tickets based on those findings.  I don't know if you want to take another look (assuming that the PR test succeeds).  Thanks!

File: test/suite/test_txn19.py [hunk 2]
```diff
@@ -346,12 +346,210 @@ def test_corrupt_log(self):
...
+        expect = list(range(0, self.nrecords))
+        salvage_config = self.base_config + ',salvage=true'
+
+        for suffix in self.suffixes:
+            self.session.create(self.uri + suffix, self.create_params)
+        self.inserts(expect)
+
+        # Simulate a crash by copying the contents of the directory
+        # before closing.  After we corrupt the copy, make another
+        # copy of the correupted directory.
```
[the remark this answers is not in the document]
Comment (ddanderson) [reply test/suite/test_txn19.py]: Fixed

File: test/suite/test_txn19.py [hunk 3]
```diff
@@ -346,12 +346,210 @@ def test_corrupt_log(self):
...
+        "garbage-end:WiredTigerLAS.wt",
+    ]
+
+    # The cases for which salvage will not work, represented in the
+    # form (self.kind + ':' + self.filename)
+    not_salvageable = [
+        "removal:WiredTiger.turtle",
+        "removal:WiredTiger.wt",
+        "truncate:WiredTiger.wt",
+        "zero:WiredTiger.wt",
```
[the remark this answers is not in the document]
Comment (ddanderson) [reply test/suite/test_txn19.py]: @kommiharibabu, actually there is a `kind='zero'` in the list (e.g. line 369).  It is different from the name of the scenario.  `kind='zero'` matches the scenarios labeled 'zero-begin' and 'zero-trunc'.

**Pull request merged**